packages/sdk/src/transactions/PSBTBuilder.ts

```diff
@@ -0,0 +1,468 @@
+
+  protected inputAmount = 0;
+
+  protected inputs: InputType[] = [];
+
+  protected outputAmount = 0;
+
+  protected outputs: Output[] = [];
+
+  protected psbt: Psbt;
+
+  protected publicKey: string;
+
+  protected rbf = true;
+
+  protected utxos: UTXOLimited[] = [];
+
+  protected usedUTXOs: string[] = [];
+
+  private autoAdjustment: boolean;
+
+  private instantTradeMode: boolean;
+
+  private nativeNetwork: networks.Network;
+
+  private noMoreUTXOS = false;
+
+  get data() {
+    return {
+      fee: this.fee,
+      virtualSize: this.virtualSize,
+      weight: this.weight,
+      changeAmount: this.changeAmount,
+      inputAmount: this.inputAmount,
+      outputAmount: this.outputAmount,
+    };
+  }
+
+  constructor({
+    address,
+    changeAddress,
+    datasource,
+    feeRate,
+    network,
+    publicKey,
+    outputs,
+    autoAdjustment = true,
+    instantTradeMode = false,
+  }: PSBTBuilderOptions) {
+    super({
+      feeRate,
+      network,
+    });
+    this.address = address;
+    this.changeAddress = changeAddress;
+    this.datasource =
+      datasource || new JsonRpcDatasource({ network: this.network });
+    this.outputs = outputs;
+    this.nativeNetwork = getNetwork(network);
+    this.publicKey = publicKey;
+
+    this.autoAdjustment = autoAdjustment;
+    this.instantTradeMode = instantTradeMode;
+
+    this.psbt = new Psbt({ network: this.nativeNetwork });
+  }
+
+  toPSBT() {
+    return this.psbt;
+  }
+
+  toHex() {
+    return this.psbt.toHex();
+  }
+
+  toBase64() {
+    return this.psbt.toBase64();
+  }
+
+  enableRBF() {
+    this.rbf = true;
+    this.addInputs();
+  }
+
+  disableRBF() {
+    this.rbf = false;
+    this.addInputs();
+  }
+
+  get xKey() {
+    return toXOnly(Buffer.from(this.publicKey, "hex")).toString("hex");
+  }
+
+  get inputsToSign() {
+    const instantTradeSellerFlow =
+      this.instantTradeMode && !this.autoAdjustment;
+    return this.psbt.txInputs.reduce(
+      (acc, _, index) => {
+        if (
+          !this.instantTradeMode ||
+          (this.instantTradeMode && index !== INSTANT_BUY_SELLER_INPUT_INDEX)
+        ) {
+          acc.signingIndexes = acc.signingIndexes.concat(index);
+        }
+
+        if (instantTradeSellerFlow) {
+          acc.sigHash =
+            // eslint-disable-next-line no-bitwise
+            Transaction.SIGHASH_SINGLE | Transaction.SIGHASH_ANYONECANPAY;
+        }
+
+        return acc;
+      },
+      {
+        address: this.address,
+        signingIndexes: [],
+      } as InputsToSign,
+    );
+  }
+
+  protected initPSBT() {
+    this.psbt = new Psbt({ network: getNetwork(this.network) }); // create new PSBT
+    this.psbt.setMaximumFeeRate(this.feeRate);
+  }
+
+  protected getInputSequence() {
+    return this.rbf ? 0xfffffffd : 0xffffffff;
+  }
+
+  private injectInput(injectable: InjectableInput) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (this.psbt.data.globalMap.unsignedTx as any).tx.ins[
```

Comment:
is this a hack? couldn't find any known interface for `unsignedTx` exposing `tx` parameter

Reply:
Yes, it's more of monkey patching approach that we have used to bind the inputs and outputs.